### PR TITLE
Functionality to retrieve auto release policy

### PR DESCRIPTION
**Pull request by JacobValdemar**

### CODE REVIEW

File: issue_release_status.go [hunk 1]
```diff
@@ -31,12 +54,85 @@ func (handler *PRCreateHandler) Handle(ctx context.Context, eventType, deliveryI
...
+
+	// Create client
+	client := &http.Client{}
+
+	// Set route
+	resp, err := client.Get(servicePath + serviceName)
+	if err != nil {
+		return errors.Wrap(err, "issueing a GET to specified URL")
+	}
+
```
Comment (Crevil) [inline issue_release_status.go:76]: This can be removed as it is the same as you do in `client.Do` in line 84 :) 

`client.Get` is a shortcut for what you do next with `NewRequest` and `client.Do`

File: issue_release_status.go [hunk 2]
```diff
@@ -31,12 +54,85 @@ func (handler *PRCreateHandler) Handle(ctx context.Context, eventType, deliveryI
...
+	req, err := http.NewRequest("GET", servicePath+serviceName, nil) // dublicate work??
+	if err != nil {
+		return errors.Wrap(err, "wrapping new request")
+	}
+
+	req.Header.Add("Authorization", "Bearer "+handler.releaseManagerAuthToken)
+
+	resp, err = client.Do(req)
```
Comment (Crevil) [inline issue_release_status.go:84]: The `err` return value is not checked here, so if the request fails, we won't know 🙈

File: issue_release_status.go [hunk 3]
```diff
@@ -31,12 +54,85 @@ func (handler *PRCreateHandler) Handle(ctx context.Context, eventType, deliveryI
...
+
+	var policyResponse ListPoliciesResponse
+
+	body, err := ioutil.ReadAll(resp.Body)
```
Comment (Crevil) [inline issue_release_status.go:88]: Again, `err` is not checked here :)

File: issue_release_status.go [hunk 4]
```diff
@@ -31,12 +54,85 @@ func (handler *PRCreateHandler) Handle(ctx context.Context, eventType, deliveryI
...
+
+	// Set route
+	resp, err := client.Get(servicePath + serviceName)
+	if err != nil {
+		return errors.Wrap(err, "issueing a GET to specified URL")
+	}
+
+	req, err := http.NewRequest("GET", servicePath+serviceName, nil) // dublicate work??
+	if err != nil {
+		return errors.Wrap(err, "wrapping new request")
```
Comment (Crevil) [inline issue_release_status.go:79]: What we actually do here, is create a request object, so the added error message might be more like the following?

```suggestion
		return errors.Wrap(err, "create GET request for release-manager")
```

File: issue_release_status.go [hunk 5]
```diff
@@ -31,12 +54,85 @@ func (handler *PRCreateHandler) Handle(ctx context.Context, eventType, deliveryI
...
+
+	resp, err = client.Do(req)
+
+	var policyResponse ListPoliciesResponse
+
+	body, err := ioutil.ReadAll(resp.Body)
+
+	defer resp.Body.Close()
+
+	json.Unmarshal(body, &policyResponse)
```
Comment (Crevil) [inline issue_release_status.go:92]: [`json.Unmarshal`](https://godoc.org/encoding/json#Unmarshal) returns an error in case of ... well errors 🙈 

It should be checked :)

File: issue_release_status.go [hunk 6]
```diff
@@ -31,12 +54,85 @@ func (handler *PRCreateHandler) Handle(ctx context.Context, eventType, deliveryI
...
+
+	var autoReleaseEnvironments []string
+	var botMessage string
+
+	// Not that easy to read, sry
+	if len(policyResponse.AutoReleases) > 0 {
```
Comment (Crevil) [inline issue_release_status.go:98]: Go has an idiom about early returns. This helps with the concept of "line of sight" by keeping the happy path to the left. (Read more here: https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88)

So we would usually invert the `if` statement:
```suggestion
	if len(policyResponse.AutoReleases) == 0 {
		log.Debug().Msg("No auto-release policies was detected for the base branch")
		return nil
	}
	// do the rest
```

This also simplifies the code and thus the comment about readability can disappear 🙈

File: issue_release_status.go [hunk 7]
```diff
@@ -31,12 +54,85 @@ func (handler *PRCreateHandler) Handle(ctx context.Context, eventType, deliveryI
...
+		for i := 0; i < len(policyResponse.AutoReleases); i++ {
+			if policyResponse.AutoReleases[i].Branch == prBase {
+				autoReleaseEnvironments = append(autoReleaseEnvironments, policyResponse.AutoReleases[i].Environment)
+			}
+		}
+
+		botMessage += "Auto-Release Policy detected for service " + serviceName + " at branch " + prBase + "\n\n"
```
Comment (Crevil) [inline issue_release_status.go:105]: You might see a case where there are auto release policies for other branches and thus the `autoReleaseEnvironments` is actually empty here.

Say I create a PR against branch `dev` but there are auto release policies for `master`. This will result in `policyResponse.AutoReleases` having items, but non of them match the base branch :) 

TL;DR: we should have a check again about policies on the base branch.

File: main.go [hunk 8]
```diff
@@ -1,23 +1,30 @@
...
 	"goji.io/pat"
 )
 
 func main() {
+	logger := zerolog.New(os.Stdout).With().Timestamp().Logger()
 
-	configPath := flag.String("config-path", "config.yml", "path to configuration file")
-	flag.Parse()
+	configPath := pflag.String("config-path", "config.yml", "path to configuration file")
+	releaseManagerAuthToken := pflag.String("release-manager-auth-token", "", "auth token for accesing release manager")
```
Comment (Crevil) [inline main.go:19]:
```suggestion
	releaseManagerAuthToken := pflag.String("release-manager-auth-token", "", "auth token for accessing release manager")
```

File: issue_release_status.go [hunk 9]
```diff
@@ -31,12 +56,95 @@ func (handler *PRCreateHandler) Handle(ctx context.Context, eventType, deliveryI
...
 	if event.GetAction() != "created" {
 		return nil
 	}
 
+	prBase := event.GetPullRequest().GetBase().GetRef() // The branch which the pull request is ending.
+
+	// Retrieve auto-release-policy
+
+	serviceName := event.GetRepo().GetName()
+	serviceName = strings.TrimSuffix(serviceName, "-service")
```
Comment (kaspernissen) [inline issue_release_status.go:68]: We could down the road consider making these configurable as well, as this is created in the open.

File: issue_release_status.go [hunk 10]
```diff
@@ -31,12 +56,95 @@ func (handler *PRCreateHandler) Handle(ctx context.Context, eventType, deliveryI
...
+	serviceName = strings.TrimPrefix(serviceName, "lunar-way-")
```
Comment (kaspernissen) [inline issue_release_status.go:69]: Same as above
